augur/align.py

```diff
@@ -2,42 +2,30 @@
 Align multiple sequences from FASTA.
 """
 
-import os,sys,argparse
+import os
 from shutil import copyfile
 import numpy as np
-from Bio import AlignIO, SeqIO, Seq
+from Bio import AlignIO, SeqIO, Seq, Align
 from .utils import run_shell_command, nthreads_value
 
-def make_gaps_ambiguous(aln):
-    '''
-    replace all gaps by 'N' in all sequences in the alignment. TreeTime will treat them
-    as fully ambiguous and replace then with the most likely state. This modifies the
-    alignment in place.
-
-    Parameters
-    ----------
-    aln : MultipleSeqAlign
-        Biopython Alignment
-    '''
-    for seq in aln:
-        seq_array = np.array(seq)
-        gaps = seq_array=='-'
-        seq_array[gaps]='N'
-        seq.seq = Seq.Seq("".join(seq_array))
-
+class AlignmentError(Exception):
+    # TODO: this exception should potentially be renamed and made augur-wide
+    # thus allowing any module to raise it and have the message printed & augur
+    # exit with code 1
+    pass
 
 def register_arguments(parser):
-    parser.add_argument('--sequences', '-s', required=True, help="sequences in fasta or VCF format")
-    parser.add_argument('--output', '-o', help="output file")
+    parser.add_argument('--sequences', '-s', required=True, nargs="+", metavar="FASTA", help="sequences to align")
+    parser.add_argument('--output', '-o', default="alignment.fasta", help="output file (default: %(default)s)")
```

Comment:
Not a bad idea to have defaults for `--output`, but strange that `augur align` is the only command to get this. Would be the plan be to add defaults to all the other `--output` arguments?

Reply:
This functionality wasn't actually added in this PR (it was set in an if/else clause at lines 57-60), this syntax is simply a nicer way to express this. Happy for the default to be removed via this PR if desired?